@here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts

```diff
@@ -79,7 +118,12 @@ export class VersionedLayerClient {
         const dataHandle = dataRequest.getDataHandle();
         const partitionId = dataRequest.getPartitionId();
         const quadKey = dataRequest.getQuadKey();
-        const version = dataRequest.getVersion();
+
+        if (!this.version) {
```

Comment:
Layer version could be 0. Use another check, please
```suggestion
        if (this.version === undefined) {
```